dashbuilder/dashbuilder-client/dashbuilder-renderers/dashbuilder-renderer-c3/src/main/java/org/dashbuilder/renderer/c3/client/charts/map/GeoData.java

```diff
@@ -0,0 +1,72 @@
+package org.dashbuilder.renderer.c3.client.charts.map;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Map.Entry;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.dashbuilder.renderer.c3.client.exports.NativeLibraryResources;
+
+import com.google.gwt.json.client.JSONArray;
+import com.google.gwt.json.client.JSONParser;
+import com.google.gwt.json.client.JSONValue;
+
+@ApplicationScoped
+public class GeoData {
+    
+    private JSONArray countries;
+
+    @PostConstruct
+    public void loadGeoJson() {
+        String jsonString = NativeLibraryResources.INSTANCE.countriesgeojson().getText();
+        countries = JSONParser.parseStrict(jsonString).isObject().get("features").isArray();
+    }
+    
+    public JSONArray countries() {
+        return countries;
+    }
+    
+    public  String countryName(JSONValue value) {
```

Comment:
This method and countryID look like implementation detail to me that should be private. If anyone calls it with some random JSONObject this will most likely fail. Wdyt?

Reply:
You are right, I will encapsulate it better and hide the JSON object details from the API.